src/rules/jsxBanPropsRule.ts

```diff
@@ -0,0 +1,81 @@
+/**
+ * @license
+ * Copyright 2017 Palantir Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as Lint from "tslint";
+import * as ts from "typescript";
+
+export class Rule extends Lint.Rules.AbstractRule {
+    /* tslint:disable:object-literal-sort-keys */
+    public static metadata: Lint.IRuleMetadata = {
+        ruleName: "jsx-ban-props",
+        description: "Bans the use of specific props.",
+        optionsDescription: Lint.Utils.dedent`
+            A list of \`['propName', 'optional explanation here']\` which bans the prop called 'propName'.`,
+        options: {
+            type: "list",
+            listType: {
+                type: "string",
+                items: {type: "string"},
+                minLength: 1,
+                maxLength: 2,
+            },
+        },
+        optionExamples: [`[true, ["someProp], ["anotherProp", "Optional explanation"]]`],
+        type: "functionality",
+        typescriptOnly: false,
+    };
+    /* tslint:enable:object-literal-sort-keys */
+
+    public static FAILURE_STRING_FACTORY = (propName: string, explanation?: string) => {
+        return `Use of the prop '${propName}' is not allowed.${explanation ? " " + explanation : ""}`;
+    }
+
+    public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
+        const options = this.getOptions();
+        const walker = new JsxBanPropsWalker(sourceFile, options);
+        const propsToBan = options.ruleArguments;
+        if (propsToBan !== undefined) {
+            propsToBan.forEach(walker.addBannedProp);
```

Comment:
can you please do this kind of options handling in the walker constructor? that would make this implementation more consistent in style with other tslint core rules.